src/org/loklak/harvester/TwitterScraper.java

```diff
@@ -520,8 +520,8 @@ public static String unshorten(String text) {
             try {
                 Matcher m = timeline_embed_pattern.matcher(text);
                 if (m.find()) {
-                    String shorturl = RedirectUnshortener.unShorten(m.group(2));
-                    text = m.replaceFirst("https://pic.twitter.com/" + shorturl + " ");
+                    String shorturl = RedirectUnshortener.unShorten(m.group(1));
+                    text = m.replaceFirst(" https://pic.twitter.com/" + shorturl + " ");

```

Comment:
The space is to prevent the case when the user uploads a photo. the pic.twitter.com link is joined with the text, this also causes the link field to be joined. For eg:
![screenshot 2016-05-22 09 34 00](https://cloud.githubusercontent.com/assets/13071508/15451634/b0cdc658-2000-11e6-9f29-84258eea5c89.png)
![screenshot 2016-05-22 09 34 07](https://cloud.githubusercontent.com/assets/13071508/15451635/b3657e6a-2000-11e6-9633-85507c50bf3e.png)
For me the text is still trimmed in the end. I'll check again and squash the commits too.